xarray/core/datatree.py

```diff
@@ -1896,3 +1898,102 @@ def apply_indexers(dataset, node_indexers):
+        **kwargs : dict
+            Additional keyword arguments passed on to ``dask.compute``.
+
+        See Also
+        --------
+        dask.compute
+        """
+        # access .data to coerce everything to numpy or dask arrays
+        lazy_data = {
+            path: {
+                k: v._data
+                for k, v in node.variables.items()
+                if is_chunked_array(v._data)
+            }
+            for path, node in self.subtree_with_keys
+        }
+        flat_lazy_data = {
+            (path, var_name): array
+            for path, node in lazy_data.items()
+            for var_name, array in node.items()
+        }
+        if lazy_data:
+            chunkmanager = get_chunked_array_type(*flat_lazy_data.values())
+
+            # evaluate all the chunked arrays simultaneously
+            evaluated_data: tuple[np.ndarray[Any, Any], ...] = chunkmanager.compute(
+                *flat_lazy_data.values(), **kwargs
+            )
+
+            for (path, var_name), data in zip(
+                flat_lazy_data, evaluated_data, strict=False
+            ):
+                self[path].variables[var_name].data = data
+
+        # load everything else sequentially
+        for node in self.subtree:
+            for k, v in node.variables.items():
+                if k not in lazy_data:
+                    v.load()
+
+        return self
+
+    def compute(self, **kwargs) -> Self:
+        """Manually trigger loading and/or computation of this datatree's data
+        from disk or a remote source into memory and return a new dataset.
+        Unlike load, the original dataset is left unaltered.
+
+        Normally, it should not be necessary to call this method in user code,
+        because all xarray functions should either work on deferred data or
+        load data automatically. However, this method can be necessary when
+        working with many file objects on disk.
+
+        Parameters
+        ----------
+        **kwargs : dict
+            Additional keyword arguments passed on to ``dask.compute``.
+
+        Returns
+        -------
+        object : DataTree
+            New object with lazy data variables and coordinates as in-memory arrays.
+
+        See Also
+        --------
+        dask.compute
+        """
+        new = self.copy(deep=False)
+        return new.load(**kwargs)
+
+    @property
+    def chunksizes(self) -> Mapping[Hashable, tuple[int, ...]]:
+        """
+        Mapping from group paths to a mapping of dimension names to block lengths for this dataset's data, or None if
+        the underlying data is not a dask array.
+
+        Cannot be modified directly, but can be modified by calling .chunk().
+
+        See Also
+        --------
+        DataTree.chunk
+        Dataset.chunksizes
+        """
+        return {
+            f"/{path}" if path != "." else "/": get_chunksizes(node.variables.values())
+            for path, node in self.subtree_with_keys
```

Comment:
does anyone know how to convert the paths to absolute (i.e. as they are displayed in the `repr`)? Or do we need to change `subtree_with_keys` to use `NodePath("/")`?

Reply:
`subtree_with_keys` should not return a `NodePath` object as that class is private (for now at least). It's also useful that it returns relative paths. We could imagine a version of that method that returns absolute paths though, Are you just asking if there is a convenience function to do this and replace this line in your PR? Or is there a problem?

Reply:
I would use either full paths or relative paths here, i.e.,
```suggestion
            node.path: get_chunksizes(node.variables.values())
            for node in self.subtree
```

or (relative paths):
```suggestion
            path: get_chunksizes(node.variables.values())
            for path, node in self.subtree_with_keys
```

I don't think we want to return paths that start with `/` that are not relative to the root of the tree.

Reply:
> Are you just asking if there is a convenience function to do this and replace this line in your PR?

Yes, I'd like to avoid having the explicit transformation everywhere

Reply:
thanks, @shoyer. My main motivation for absolute paths is that that's what people see in the `repr`, and thus might be easier to interpret. I'll switch to `tree.subtree`.